File: java/src/IceSSL/src/main/java/com/zeroc/IceSSL/ConnectorI.java
// **********************************************************************
//
// Copyright (c) 2003-2018 ZeroC, Inc. All rights reserved.
//
// This copy of Ice is licensed to you under the terms described in the
// ICE_LICENSE file included in this distribution.
//
// **********************************************************************

package com.zeroc.IceSSL;

import com.zeroc.IceInternal.Network;

final class ConnectorI implements com.zeroc.IceInternal.Connector
{
    @Override
    public com.zeroc.IceInternal.Transceiver connect()
    {
        //
        // The plug-in may not be fully initialized.
        //
        if(!_instance.initialized())
        {
            com.zeroc.Ice.PluginInitializationException ex = new com.zeroc.Ice.PluginInitializationException();
            ex.reason = "IceSSL: plug-in is not initialized";
            throw ex;
        }

        return new TransceiverI(_instance, _delegate.connect(), _host, false);
    }

    @Override
    public short type()
    {
        return _delegate.type();
    }

    @Override
    public String toString()
    {
        return _delegate.toString();
    }

    @Override
    public int hashCode()
    {
        return _delegate.hashCode();
    }

    //
    // Only for use by EndpointI.
    //
    ConnectorI(Instance instance, com.zeroc.IceInternal.Connector delegate, String host)
    {
        _instance = instance;
        _delegate = delegate;
        _host = host;
    }

    @Override
    public boolean equals(java.lang.Object obj)
    {
        if(!(obj instanceof ConnectorI))
        {
            return false;
        }

        if(this == obj)
        {
            return true;
        }

        ConnectorI p = (ConnectorI)obj;
        return p._delegate.equals(_delegate);
    }

    private Instance _instance;
    private com.zeroc.IceInternal.Connector _delegate;
    private String _host;
}

File: csharp/src/Ice/IceSSL/ConnectorI.cs
// Copyright (c) ZeroC, Inc.

namespace IceSSL;

internal sealed class ConnectorI : IceInternal.Connector
{
    public IceInternal.Transceiver connect() =>
        new TransceiverI(
            _instance,
            _delegate.connect(),
            _host,
            incoming: false,
            serverAuthenticationOptions: null);

    public short type() => _delegate.type();

    // Only for use by EndpointI.
    internal ConnectorI(Instance instance, IceInternal.Connector del, string host)
    {
        _instance = instance;
        _delegate = del;
        _host = host;
    }

    public override bool Equals(object obj)
    {
        if (!(obj is ConnectorI))
        {
            return false;
        }

        if (this == obj)
        {
            return true;
        }

        var p = (ConnectorI)obj;
        return _delegate.Equals(p._delegate);
    }

    public override string ToString() => _delegate.ToString();

    public override int GetHashCode() => _delegate.GetHashCode();

    private readonly IceInternal.Connector _delegate;
    private readonly string _host;
    private readonly Instance _instance;
}
